File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Morocco.m
//
//  UIImage+Morocco.m
//  CountriesFlags
//

#import "Countries/UIImage+Morocco.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Morocco)

+ (UIImage * _Nullable)flagMoroccoWithSize:(CGSize)size {
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:99.0f/255.0f
                                           blue:65.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:183.0f/255.0f
                                        green:49.0f/255.0f
                                         blue:44.0f/255.0f
                                        alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setStroke];
    UIBezierPath* path = [UIBezierPath starWithSize:size
                                             radius:size.height / 4.7f
                                               type:FlagStarTypeFivePointed];
    path.lineWidth = size.height / 4.0f * 0.095105652;
    [path stroke];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Morocco.swift
//
//  UIImage+Morocco.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Morocco "wikipedia"
     
     Generates the flag of Morocco.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Morocco.
     */
    internal static func flagMorocco(with size: CGSize) -> UIImage? {
        let colorGreen = UIColor(red: 0.0,
                                 green: 99.0 / 255.0,
                                 blue: 65.0 / 255.0,
                                 alpha: 1.0)
        let colorRed = UIColor(red: 183.0 / 255.0,
                               green: 49.0 / 255.0,
                               blue: 44.0 / 255.0,
                               alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setStroke()
        let path = UIBezierPath.star(at: size.midPoint, radius: size.height / 4.7)
        path.lineWidth = size.height / 4.0 * 0.095105652
        path.stroke()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
